src/main/java/org/jabref/model/openoffice/util/OOResult.java

```diff
@@ -0,0 +1,128 @@
+package org.jabref.model.openoffice.util;
+
+import java.util.Optional;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+/*
+ * error cannot be null
+ * result cannot be null
+ *
+ * Void is not allowed for R, use OOVoidResult instead.
+ *
+ * Out of `isPresent()` and `isError()` exactly one is true.
+ */
+public class OOResult<R, E> {
+    private final Optional<R> result;
+    private final Optional<E> error;
+
+    /**
+     * Exactly one of the arguments should be Optional.empty()
+     *
+     * @param result
+     * @param error
+     */
+    private OOResult(Optional<R> result, Optional<E> error) {
+        this.result = result;
+        this.error = error;
+    }
+
+    /**
+     * @param result Null is not allowed.
+     */
+    public static <R, E> OOResult<R, E> ok(R result) {
+        return new OOResult<>(Optional.of(result), Optional.empty());
+    }
+
+    /**
+     * @param error Null is not allowed.
```

Comment:
You can remove this comment. In nearly 99% of the cases, `null` is not allowed. If you really mean `Null`, please reference that class by using JavaDoc's `{@link ...}.

Reply:
removed